packages/jasmine-runner/src/jasmine-runner.ts

```diff
@@ -97,43 +83,89 @@ class JasmineRunner implements TestRunner {
+            const executionResult = await runTask.promise;
+            Util.handleDuplicateTests(executionResult.testResults);
             if (locators.length > 0) {
-                executionResults.testResults = Util.filterTestResultsByTestLocator(executionResults.testResults,
+                executionResult.testResults = Util.filterTestResultsByTestLocator(executionResult.testResults,
                     testLocators, blockListedLocators)
-                if (executionResults.testSuiteResults.length > 0) {
-                    executionResults.testSuiteResults = Util.filterTestSuiteResults(executionResults.testResults,
-                        executionResults.testSuiteResults)
+                if (executionResult.testSuiteResults.length > 0) {
+                    executionResult.testSuiteResults = Util.filterTestSuiteResults(executionResult.testResults,
+                        executionResult.testSuiteResults)
                 }
             }
-            if (postTestResultsEndpoint) {
-                await Util.makeApiRequestPost(postTestResultsEndpoint, executionResults);
+
+            
+            for (const spec of testFilesToProcessList) 
+            {
+                //if (path.includes(spec) )
+                {
+                    console.log("Clearing spec from require cache: " + spec);
+                    delete require.cache[spec];
+                }
             }
-            return executionResults;
+            return executionResult;
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
         } catch (err: any) {
             throw new RunnerException(err.stack);
         }
+
+    }
+    async executeTests(argv: parser.Arguments): Promise<ExecutionResults> {
+        const taskID = process.env.TASK_ID as ID;
+        const buildID = process.env.BUILD_ID as ID;
+        const orgID = process.env.ORG_ID as ID;
+        const repoID = process.env.REPO_ID as ID;
+        const commitID = process.env.COMMIT_ID as ID;
+        Validations.validateExecutionEnv(argv);
+        const postTestResultsEndpoint = process.env.ENDPOINT_POST_TEST_RESULTS as string || "";
+        const testFilesGlob = argv.pattern as string | string[];
+        const locatorFile = argv.locatorFile as string;
+        let locators: InputConfig = new InputConfig();
+        if (locatorFile) {
+            locators = Util.getLocatorsConfigFromFile(locatorFile)
+        }
+        const executionResults = new ExecutionResults(
+            taskID,
+            buildID,
+            repoID,
+            commitID,
+            orgID,
+            []
+        );
+        const locatorSet = Util.createLocatorSet(locators)
+        for (const set of locatorSet) {
+            for (let i=1; i<=set.n; i++) {
+                const result = await this.execute(testFilesGlob, set.locators, argv.config)
+                executionResults.push(result)
+            }
+        }
```

Comment:
We used to run all the tests if `--locatorFile` wasn't provided. I think, after this change we're taking away that capability and even for local runs, one need to provide a locatorFile. This takes away the general usability of this runner as a library.

Reply:
added support in all 3 runners